File: portScanner/portScanner.py
import socket
import sys


if len(sys.argv) == 1:
    print("please specify host to scan")
    sys.exit(1)
host = sys.argv[1]

print(f"scanning {host}")
try:
    for port in range(1, 65535):
        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        sock.settimeout(1)
        connection = sock.connect_ex(("1.0.0.1", port))
        if connection == 0 :
            print(f"{port} is open")
        sock.close()
except Exception as err:
    print(err)

File: portScanner/portScanner.go
package main

import (
    "net"
    "fmt"
    "os"
)

func main() {
    if len(os.Args) == 1 {
        fmt.Println("Please specify a host\nportScanner <host>")
        os.Exit(1)
    }
    host := os.Args[1]
    startPort := 1
    endPort := 65535
    for ;startPort <= endPort; startPort++ {
        connection, err := net.Dial("tcp", fmt.Sprintf("%v:%v", host,startPort))
        if err != nil {
            continue
        }
        fmt.Printf("%v is open\n", startPort)
        connection.Close()
    }

}
